### ISSUE TITLE
[Bug report] Name '...' is not defined when wildcard importing local module

### ISSUE THREAD

**Issue by ErwanDL** [python/mypy#8497]
Mypy version : 0.761

Mypy outputs an error `Name 'MY_CONSTANT' is not defined` when `MY_CONSTANT` has been imported through a wildcard import of a local module. The error doesn't appear when I wildcard import a standard library module for instance (such as `from math import *`.

This is definitely a false positive, since I can run `main.py` without any error.

Example code : 
```python
# main.py
from mypackage.my_test_module import *

print(MY_CONSTANT)
```

The module in question : 
```python
# my_test_module.py
MY_CONSTANT = "Hello world"
```

Simplified project architecture : 
```
mypackage
┣ my_test_module.py
┗ main.py
```

**Comment by ilevkivskyi** [comment]
Most likely you are using `--ignore-missing imports`, and since you don't have an `__init__.py` file there mypy doesn't find that file.

Also I have no idea how do you run Python, but without that `__init__.py` file I get
```
Traceback (most recent call last):
  File "main.py", line 1, in <module>
    from mypackage.my_test_module import *
ModuleNotFoundError: No module named 'mypackage'
```

**Comment by ErwanDL** [comment]
There is actually an `__init__.py` file, I tried to give a minimal reproducible example (the real life project on which this bug happens to me is much more complex) and didn't think it was necessary, my bad. I'll try to be more comprehensive.

Note : I do not run mypy with the `--ignore-missing imports` flag.

As you can see on the image below, I import a bunch of constants (one of them is called `WORK_DIR`) with the first line :
```
from feedly.ml_ner_business.globals import *
```
The import succeeds, and as you can see the Python extension in VSCode even recognizes the type of `WORK_DIR` (i.e. an instance of `Path`). However, mypy displays the following error : 
```
Name 'WORK_DIR' is not defined mypy(error)
```
Pylint doesn't detect any error here, neither does the Python extension, and this module works without a problem at runtime, so I can assure you that this seems to be a false positive on mypy's side.

![Screen Shot 2020-03-06 at 3 42 54 PM](https://user-images.githubusercontent.com/35762435/76130874-2c9c7780-5fc1-11ea-962f-9353611b5dec.png)

**Comment by ilevkivskyi** [comment]
> I can try and create an actual minimal reproducible project this weekend so that it is clearer where the bug may originate from, if you want.

Yes, we can't really proceed without this.

**Comment by ErwanDL** [comment]
@ilevkivskyi I have created a [basic project to reproduce this error](https://github.com/ErwanDL/mypy_wildcard_import_issue). You can create a virtualenv and install the requirements from the `requirements.txt` file.

I have noticed the following behaviour : 
- If you add an empty `__init__.py` file in _my_package_/ both wildcard and regular named imports seem to work normally.
- If you do not add an empty `__init__.py` file (which is not required anymore in Python 3 if I'm not mistaken), the variable imported via a regular named import is correctly resolved by mypy, but the variable imported via the wildcard import raises the error I mentioned above. You can observe this behaviour in the following screenshot : 

<img width="1073" alt="Screen Shot 2020-03-06 at 11 32 14 PM" src="https://user-images.githubusercontent.com/35762435/76139169-bae51d80-6002-11ea-88ac-5ac92102f91e.png">

Note that in both cases described above, running the `main.py` file with python works fine.
Is this expected behaviour ? It seems inconsistent to me that wildcard imports rely on the presence of an `__init__.py` file, whereas regular imports do not.

**Comment by ErwanDL** [comment]
@ilevkivskyi I have managed to create [a minimal project](https://github.com/ErwanDL/mypy_wildcard_issue) that reproduces the original error, and may have an idea for what causes it.
You can see that running `main.py` works fine, but mypy raises the error I described :
<img width="1127" alt="Screen Shot 2020-03-07 at 8 30 13 PM" src="https://user-images.githubusercontent.com/35762435/76156672-7913b000-60b2-11ea-9761-38d539caca29.png">

It seems that the error only appears when the path to the module that you are trying to wildcard-import has a certain level of nesting (in our case, 3 levels : `mypackage` then `mysubpackage` then `constants`). If you move the `constants.py` file one level up, directly at the root of `mypackage`, you will see that mypy doesn't raise the error anymore in `main.py`.
